File: Desktop/Nexus/frontend/src/types/market.ts
export interface Product {
  id: number
  seller_id: number
  title: string
  description: string
  category: string
  price: number
  currency: string
  cover_url?: string | null
  demo_url?: string | null
  tags: string
  rating: number
  sales: number
  is_published: boolean
  created_at: string
  updated_at: string
}

export interface ProductCreatePayload {
  title: string
  description: string
  category: string
  price: number
  currency?: string
  cover_url?: string
  demo_url?: string
  tags?: string
}

export interface Review {
  id: number
  product_id: number
  user_id: number
  rating: number
  comment?: string | null
  created_at: string
}

export interface Order {
  id: number
  product_id: number
  buyer_id: number
  amount: number
  currency: string
  status: string
  created_at: string
}

File: Desktop/Nexus/backend/app/schemas/market.py
"""Marketplace schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProductBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: str = Field(..., min_length=1)
    category: str = "template"
    price: float = Field(..., ge=0)
    currency: str = "USD"
    cover_url: str | None = None
    demo_url: str | None = None
    tags: str = ""


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    category: str | None = None
    price: float | None = None
    currency: str | None = None
    cover_url: str | None = None
    demo_url: str | None = None
    tags: str | None = None
    is_published: bool | None = None


class ProductOut(ProductBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    seller_id: int
    rating: float
    sales: int
    is_published: bool
    created_at: datetime
    updated_at: datetime


class ReviewCreate(BaseModel):
    rating: int = Field(..., ge=1, le=5)
    comment: str | None = None


class ReviewOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    user_id: int
    rating: int
    comment: str | None = None
    created_at: datetime


class OrderCreate(BaseModel):
    product_id: int


class OrderOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    buyer_id: int
    amount: float
    currency: str
    status: str
    created_at: datetime
